components/exercises/chapter-4-oscar/SumOfARange.tsx

```diff
@@ -0,0 +1,211 @@
+import Image from 'next/image';
+import React, { useState } from 'react';
+
+/**
+ * Calculates the sum of a range of numbers.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @returns An array containing all the numbers in the range.
+ */
+export const calcRangeSum = (start: number, end: number) => {
+  const range = [];
+  for (let i = start; i <= end; i++) {
+    range.push(i);
+  }
+  return range;
+};
+
+/**
+ * Calculates the sum of a range of numbers based on a given start, end, and step.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @param step - The step to increment or decrement the range by.
+ * @returns An array of numbers representing the range.
+ */
+export const calcRangeSumMod = (start: number, end: number, step: number) => {
+  const range = [];
+
+  if (step === 1) {
+    // if step is 1, then we can just use the original function
+    for (let i = start; i <= end; i += step) {
+      range.push(i);
+    }
+  } else if (step < 1) {
+    // if step is less than 1, then we need to decrement the range
+    for (let i = start; i >= end; i += step) {
+      range.push(i);
+    }
+  } else {
+    return;
+  }
+  return range;
+};
+
+/**
+ * Calculates the sum of all numbers in an array.
+ * @param arr - The array of numbers to sum.
+ * @returns The sum of all numbers in the array.
+ */
+export const arraySum = (arr: number[]) => {
+  let sum = 0;
+  for (let i = 0; i < arr.length; i++) sum += arr[i];
+  return sum;
+};
+
+const animals = [
+  { id: 1, name: 'Bear', img: '/bear.png' },
+  { id: 2, name: 'Boar', img: '/boar.png' },
+  { id: 3, name: 'Buffalo', img: '/buffalo.png' },
+  { id: 4, name: 'Monkey', img: '/monkey.png' },
+  { id: 5, name: 'Elephant', img: '/elephant.png' },
+  { id: 6, name: 'Hippo', img: '/hippo.png' },
+  { id: 7, name: 'Lion', img: '/lion.png' },
+  { id: 8, name: 'Lynx', img: '/lynx.png' },
+  { id: 9, name: 'Ostrich', img: '/ostrich.png' },
+  { id: 10, name: 'Wolf', img: '/wolf.png' },
+];
+
+const initialAnimals = {
+  start: 1,
+  end: 10,
+  step: 1,
+};
+
+export default function SumOfARange() {
+  const [showAnimals, setShowAnimals] = useState(initialAnimals);
```

Comment:
This aren't animals, it is a range (used to render animals)
const [range, setRange] = useState(initialRange);
```

Reply:
refactored